db/Pos/DB/Class.hs

```diff
@@ -95,7 +95,7 @@ type SerializedBlock = Serialized SerBlock
 type SerializedUndo = Serialized SerUndo
 
 -- | Pure read-only interface to the database.
-class (HasConfiguration, MonadThrow m) => MonadDBRead m where
```

Comment:
Huh, it seems very strange to me that someone would ever add a nullary constraint like `HasConfiguration` to a _class_, rather than to its instances. It doesn't constrain the set of instances at all; all it does is act like an obfuscated constraint synonym.